flaky/_flaky_plugin.py

```diff
@@ -233,7 +233,11 @@ def _add_flaky_report(self, stream):
             `file`
         """
         stream.write('===Flaky Test Report===\n\n')
-        stream.write(self._stream.getvalue())
+        value = self._stream.getvalue()
+        try:
+            stream.write(value)
+        except UnicodeEncodeError:
+            stream.write(ensure_byte_string(value))
```

Comment:
I've kept this as-is because it looks like Python 2 will write to the stderr stream as a byte string while Python 3 will write as text, so we will first try the write, and only call ensure_byte_string() if it fails with the UnicodeEncodeError. I've added a comment to reflect this.